lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
No need to continue this discussion @zeroedin-bill -- I had to explain to @deeky666 and @riccardonar that the suggested use of PGSQL_LIBPQ_VERSION is not a valid solution. They both disagreed so I had to prove why it's not a valid solution, hehe.

SO, at this point, the best solution would be to try to connect using client_encoding -- catch exception and detect client_encoding rejection. Reattempt connection without client_encoding and use SET NAMES instead if so.

That way -- for any recent installs, client_encoding will work as expected (and pgbouncer will be happy) -- and we'll retain support for older systems (with the SLIGHT drawback that SET NAMES is called after connection.

@deeky666 at this point -- you agree right?